rwengine/src/ai/CharacterController.cpp

```diff
@@ -239,6 +239,28 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
         } else if (character->animator->getAnimation(AnimIndexAction) ==
                    anm_enter) {
             if (character->animator->isCompleted(AnimIndexAction)) {
+                if (vehicle->getVehicle()->vehiclename_ == "TAXI"
+                    || vehicle->getVehicle()->vehiclename_ == "CABBIE"
+                    || vehicle->getVehicle()->vehiclename_ == "BORGNINE") {
+                    // Earn $25 from taxi cab
+                    // @todo
+                } else if (vehicle->getVehicle()->vehiclename_ == "POLICAR") {
+                    // Police cruisers give 5 shotgun cartridges
+                    character->addToInventory(4, 5);
```

Comment:
The ``GameState::playerObject`` value should match ``getGameObjectID()`` called on the player's character, we can just wrap that in a ``CharacterObject::isPlayer()`` function if the bonuses should only apply to the player.